Feature/ch673/create external adapter for sportsdataio

GMSteuart opened the pull request:
~~Note: leaving out updates for npm dependencies until after initial review so viewing file changes is easier.~~
Added for CI

christianagnew commented:
1) Not sure if this has changed but in the past CL said to keep API keys inside the endpoint files and out of config.ts so they dont get logged

2) README should be updated

GMSteuart commented:
@christianagnew 

I'm taking point 1 has changed since then as I extended by adding one key/value pair where the other two were already there. I will say I didn't see any API keys in the many times I looked at logs while debugging.

As for 2, I added all the of the documentation around functionality I added.

I added documentation on curling so that 1) there's more examples out there to grab from and 2) because the test cases are still using the default.
